### PR TITLE
Updates promote actions to handle multiple tracks

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: .github/workflows/charms.promote.yaml [hunk 1]
```diff
@@ -104,7 +104,8 @@ jobs:
             echo "promote-from=candidate" >> ${GITHUB_ENV}
             echo "promote-to=stable" >> ${GITHUB_ENV}
```
Comment (Gmerold) [inline .github/workflows/charms.promote.yaml:105]: Shouldn't this be `${{ github.ref_name }}/stable`?
Same applies to all `promote-to`s

File: .github/workflows/charms.promote.yaml [hunk 2]
```diff
@@ -104,7 +104,8 @@ jobs:
             echo "promote-from=candidate" >> ${GITHUB_ENV}
```
Comment (Gmerold) [inline .github/workflows/charms.promote.yaml]: Shouldn't this be `${{ github.ref_name }}/candidate`?
Same applies to all `promote-from`s

File: .github/workflows/charms.promote.yaml (hunk 1 above)
Comment (Gmerold) [reply .github/workflows/charms.promote.yaml:105]: Nvm, I'm stupid :)

File: .github/workflows/charms.promote.yaml (hunk 2 above)
Comment (Gmerold) [reply .github/workflows/charms.promote.yaml]: Nvm, I'm stupid :)

**Review by Gmerold — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**